core/src/clusterization/measurement_creation.cpp

```diff
@@ -48,7 +50,10 @@ measurement_creation::output_type measurement_creation::operator()(
         const auto &module = modules.at(module_link);
 
         // Fill measurement from cluster
-        detail::fill_measurement(result, cluster, module, module_link);
+        // Give each measurement a unique ID, for the ambiguity resolution
+        // algorithm
+        detail::fill_measurement(result, cluster, module, module_link,
+                                 ++unique_measurement_id);
```

Comment:
:thinking: Should the IDs start from `0` (my preference) or `1` (what the code does currently)? To start from zero, you'd need:

```suggestion
                                 unique_measurement_id++);
```

Reply:
Thanks for your review!

Good luck for making the GPU chain work next week. I will try to make an implementation of the greedy ambiguity solver with SYCL in the coming weeks, but I'm not sure it'll be efficient, even parallelized with small clusters.

Reply:
Using `.size()` as you mentioned solved this issue. I was a bit worried by the `measurement_id` 0, since it can be confused with an uninitialized value, but it doesn't really matter, I'll implement an additional check on the greedy solver, to warn the user if there are too many zeroes as input IDs.